### PR TITLE
Add documentation about breaking java time changes

### PR DESCRIPTION
**Pull request by spinscale**
This adds documentation for changes regarding the joda/java time switch.

### CODE REVIEW

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 1]
```diff
@@ -0,0 +1,47 @@
+[float]
+[[breaking_70_java_time_changes]]
+=== Replacing joda time with java time
+
+Since Java 8 there is a dedicated `java.time` package, which is superior to
+the joda time library, that has been used so far in Elasticsearch. One of
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: Nit: joda time -> Joda-Time (according to https://www.joda.org/joda-time/) - I won't point this out for the rest of the doc.

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+The mappings, aggregations and search code switched from joda time to
+java time.
+
+[float]
+==== Joda based date formatters are replaced with java ones
+
+With the release of Elasticsearch 6.7 a backwards compatibility layer was
+introduced, that checked if you are using a joda time based formatter, that is
+supported differently in java time. A log message was emitted, and you could
+create the proper java time based formatter prefixed with an `8`.
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: Maybe a small example is warranted? Wdyt?

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+With Elasticsearch 7.0 all formatters are now java based, which means you might
+get exceptions when keeping your old formatters the same without checking the
+depreciation log in 6.7.
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: Typo: depreciation -> deprecation

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+[float]
+==== Joda based date formatters are replaced with java ones
+
+With the release of Elasticsearch 6.7 a backwards compatibility layer was
+introduced, that checked if you are using a joda time based formatter, that is
+supported differently in java time. A log message was emitted, and you could
+create the proper java time based formatter prefixed with an `8`.
+
+With Elasticsearch 7.0 all formatters are now java based, which means you might
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: How about: "you will get exceptions ... when using deprecated formatters"?

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+get exceptions when keeping your old formatters the same without checking the
+depreciation log in 6.7.
+
+[float]
+==== Using time zones in the Java client
+
+Timezones have to specified as java time based zone objects.
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: What does this mean specifically?

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+[float]
+==== Parsing aggregation buckets in the Java client
+
+The date based aggregation buckets in responses used to be of
+type `JodaTime`. Due to moving over to java-time, the buckets are now of
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: moving over -> migrating?

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+type `ZonedDateTime`. As the client is returning untyped objects here, you
+may run into class cast exceptions only when running the code, but not in
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: but not *at* compile time?

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+compile time, ensure you have proper test coverage for this in your
+own code.
+
+[float]
+==== Parsing `GMT0` timezone with JDK8 is not supported
+
+When you are running Elasticsearch 7 under Java, you are not able to parse
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: When you are running Elasticsearch 7 with Java **8**?

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+the timezone `GMT0` properly anymore. The reason for this is a bug in the
+JDK, which has not been fixed for JDK8. You can read more information in the
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: Remove "information"?

**Review by danielmitterdorfer — COMMENTED** [review]
I think it makes sense to document those changes in a separate file. I left a few suggestions.

**Comment by spinscale** [comment]
took care of all your comments. I also grepped through the whole documentation and removed any mentions of joda time or joda specific formats in another commit within this PR

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 10]
```diff
@@ -1,47 +1,113 @@
...
+`y` in java time.
+
+The week-based-year formatting character is a lowercase `x` in Joda-Time,
+but an upper-case `Y` in java time.
 
 [float]
 ==== Using time zones in the Java client
 
-Timezones have to specified as java time based zone objects.
+Timezones have to specified as java time based zone objects. This means,
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: nit: "have to *be* specified"

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 11]
```diff
@@ -1,47 +1,113 @@
...
+* `getMinuteOfDay()` should be replaced with `get(ChronoField.MINUTE_OF_DAY)`
+* `getMinuteOfHour()` should be replaced with `getMinute()`
+* `getMonthOfYear()` should be replaced with `getMonthValue()`
+* `getSecondOfDay()` should be replaced with `get(ChronoField.SECOND_OF_DAY)`
+* `getSecondOfMinute()` should be replaced with `getSecond()`
+* `getWeekOfWeekyear()` should be replaced with `get(WeekFields.ISO.weekOfWeekBasedYear())`
+* `getWeekyear()` should be replaced with `get(WeekFields.ISO.weekBasedYear())`
+* `getYearOfCentury()` should be replaced with `get(ChronoField.YEAR_OF_ERA) % 100`
+* `getYearOfEra()` should be replaced with `get(ChronoField.YEAR_OF_ERA)`
+* `toString(String)` should be replaced with `a DateTimeFormatter`
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: nit: "`a DateTimeFormatter`" -> "a `DateTimeFormatter`"?

File: docs/reference/migration/migrate_7_0/java_time.asciidoc [hunk 12]
```diff
@@ -1,47 +1,113 @@
...
+* `toString(String,Locale)` should be replaced with `a DateTimeFormatter`
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/java_time.asciidoc]: nit: "`a DateTimeFormatter`" -> "a `DateTimeFormatter`"?

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for iterating, this looks great. I left only very minor comments (which won't require another review round). LGTM

**Comment by rjernst** [comment]
@spinscale This should probably be partially backported to 6.7?

**Pull request merged**